### PR TITLE
Vault integration with AWS Lambda and fix security vulnerabilities. Signed-off-by: Anupreksha Jain <janupreksha@vmware.com>

### PR DESCRIPTION
**Pull request by ajain0811**

### CODE REVIEW

File: aws/src/index.js [hunk 1]
```diff
@@ -819,35 +822,67 @@ const handleRecords = (event, context, lintEnv) => {
...
+
+            if (event.Records || event.records) {
+                handleRecords(event, context, lintEnv);
+            }
+
+            if (process.env.EventBridge_Logs === 'true') {
+                handleEventBridgelogs(event, context, lintEnv);
+            }
+        }).catch(function(err) {
+            console.log(err);
```
Comment (slk244) [inline aws/src/index.js:862]: should you use handleError here as well?

File: aws/src/vault.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const http = require('http');
```
Comment (slk244) [inline aws/src/vault.js:1]: if possible lets try to write a test case for these new workflows

File: aws/src/index.js [hunk 3]
```diff
@@ -819,35 +822,67 @@ const handleRecords = (event, context, lintEnv) => {
...
+        if (!apiToken) {
+            handleError('The API token is missing. Please configure it in an environment variable of the lambda function');
+            return;
+        }
+        lintEnv = new LIntHttpEnv(`Bearer ${apiToken}`, ingestionUrl);
 
-  if (event.Records || event.records) {
-    handleRecords(event, context, lintEnv);
-  }
+        if (event.awslogs) {
```
Comment (slk244) [inline aws/src/index.js:872]: why is this change done now? did we see any issues in the previous code?

File: aws/src/index.js (hunk 1 above)
Comment (ajain0811) [reply aws/src/index.js]: Incorporated the change.

File: aws/src/vault.js (hunk 2 above)
Comment (ajain0811) [reply aws/src/vault.js:1]: Incorporated the change.

File: aws/src/index.js (hunk 3 above)
Comment (ajain0811) [reply aws/src/index.js:871]: just provided the condition on behalf of vault integration, Didn't modify the previous code.

**Pull request merged**